Source/Client.swift

```diff
@@ -49,29 +49,28 @@ extension Client {
         request.host = "\(host):\(port)"
         request.userAgent = "Zewo"
 
-        if request.connection == nil {
+        if request.connection == [] {
             request.connection = "close"
         }
 
         return request
     }
 
-    public func respond(request: Request) throws -> Response {
+    public func respond(to request: Request) throws -> Response {
         let request = addHeaders(request)
-        let stream = try client.connect()
+        try self.connection.open(timingOut: .never)
+        try serializer.serialize(request, to: connection)
 
-        try serializer.serialize(request) { data in
-            try stream.send(data)
-        }
-
-        try stream.flush()
+        try connection.flush()
 
         while true {
-            let data = try stream.receive()
+            var data = Data()
+            for chunk in StreamSequence(for: connection) {
+                data.bytes.append(contentsOf: chunk.bytes)
+            }
```

Comment:
While it isn't beautiful, the `reduce` @dreymonde is actually quadratic in time complexity (bytes array is copied every iteration). The for loop is a better approach.

A slightly more pretty alternative::
```swift
let data: Data = {
    var data = Data()
    for chunk in StreamSequence(for: connection) {
        data.bytes.append(contentsOf: chunk.bytes)
    }
    return data
}()
```